### PR TITLE
Woo: add methods to update a product

### LINKED ISSUE #1431: Woo: support updating a product
The existing FluxC code currently includes the endpoint needed to update a product: This is done via PUT `/wp-json/wc/v3/products/<id>`, but we still need separate actions/methods to update product info.

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #1431. This PR adds the ability to update a product description, name + sku. 

### Changes
- Adds new actions for updating a product: `UPDATE_PRODUCT` and `UPDATED_PRODUCT`.
- Adds new payload classes for updating a product: `UpdateProductPayload`, `RemoteUpdateProductPayload`.
- Adds a new fragment to the example app to test updating a product description, name and sku.
- Adds new tests to `MockedStack_WCProductsTest` and `ReleaseStack_WCProductTest`.

### Screenshots
 .

### CODE REVIEW

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCProductStore.kt [hunk 1]
```diff
@@ -290,6 +315,8 @@ class WCProductStore @Inject constructor(dispatcher: Dispatcher, private val wcP
                 updateProductReviewStatus(action.payload as UpdateProductReviewStatusPayload)
             WCProductAction.UPDATE_PRODUCT_IMAGES ->
                 updateProductImages(action.payload as UpdateProductImagesPayload)
+            WCProductAction.UPDATE_PRODUCT ->
```
Comment (nbradbury) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCProductStore.kt]: We can do this in a separate PR, but I'm wondering if we should drop `UPDATE_PRODUCT_IMAGES` and rely solely on `UPDATE_PRODUCT`?

Comment (anitaa1990) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCProductStore.kt]: I think that's a good idea, especially since we would now be updating images along side all other items in a product only when **publish/update** is initiated from the UI.
